src/render/src/buffer.rs

```diff
@@ -17,5 +19,64 @@ pub struct Info {
     /// Stride of a single element, in bytes. Only used for structured buffers
     /// that you use via shader resource / unordered access views.
     pub stride: u64,
-    // TODO: mapping stuff
+    /// Exclusive access
+    pub(crate) access: Access,
+}
+
+impl Info {
+    pub(crate) fn new(usage: Usage, memory: Memory, size: u64, stride: u64)
+        -> Self
+    {
+        let access = Access {
+            cpu: AtomicBool::new(false),
+            gpu: AtomicUsize::new(0),
+        };
+        Info { usage, memory, size, stride, access }
+    }
+}
+
+#[derive(Debug)]
+pub(crate) struct Access {
+    cpu: AtomicBool,
```

Comment:
I wonder if we could use `Arc<()>` for that or something, to reduce the atomic logic

Reply:
`AtomicBool` can probably be replaced by `Mutex<()>` but then you have to drag the `MutexGuard`s everywhere and the lifetimes will probably not be manageable. Not sure how you would use `Arc<()>`.